shell/server/pre-meteor.js

```diff
@@ -43,6 +43,12 @@ function isSandstormShell(hostname) {
   return (hostname === HOSTNAME || (DDP_HOSTNAME && hostname === DDP_HOSTNAME));
 }
 
+function hasPseudoTLD(hostname) {
+  // Returns true if this hostname ends in one of the major pseudo-TLDs.
+  let tld = hostname.split(".").pop();
+  return (tld === ("onion" || "i2p" || "gnu"));
```

Comment:
In my testing, this seems to use short-circuit evaluation of `||`.

I think this would be cleaner if the function has a list called

```
var pseudoTlds = ['onion', 'i2p', 'gnu'];
```

and then you check for membership by e.g.

`pseudoTlds.indexOf(tld) !== -1`

I forget how to do stuff like this all the time in Javascript, so here is my interactive session where I verified the above:

```
➜  ~  js
> 'x' || 'y'
'x'
> var x = ['a', 'b'];
undefined
> x.indexOf('a');
0
> x.indexOf('z');
-1
```